### PR TITLE
Rmtree

### PR DESCRIPTION
**Pull request by CJ-Wright**
Add rmtree to lib (works properly on windows)
@scopatz should this auto force?

### CODE REVIEW

File: news/rmtree.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
+**Added:**
+
+* ``xonsh.lib.rmtree`` an rmtree which works on windows properly (even with
```
Comment (scopatz) [inline news/rmtree.rst]: Shouldn't this be `xonsh.lib.os.rmtree()`

File: xonsh/lib/os.xsh [hunk 2]
```diff
@@ -9,3 +9,23 @@ def indir(d):
...
+
+    Parameters
+    ----------
+    dirname : str
+        Directory to be removed
+    """
+    try:
+        rm -r @(dirname)
+    except PermissionError:
+        if sys.platform == "win32":
```
Comment (scopatz) [inline xonsh/lib/os.xsh]: I think you need to import sys

File: xonsh/lib/os.xsh [hunk 3]
```diff
@@ -9,3 +9,23 @@ def indir(d):
...
+            from xonsh.lib import subprocess
+            subprocess.check_call(["del", "/F/S/Q", dirname])
```
Comment (scopatz) [inline xonsh/lib/os.xsh]: Also, couldn't you just do `del /F/S/Q @(dirname)`

**Comment by scopatz** [comment]
Hmm it probably shouldn't autoforce, but that would be a nice option.

**Comment by scopatz** [comment]
Thanks for putting this in!

**Comment by CJ-Wright** [comment]
@scopatz do you want a squash of some of these?

File: news/rmtree.rst [hunk 4]
```diff
@@ -0,0 +1,14 @@
+**Added:**
+
+* ``xonsh.lib.rmtree()`` an rmtree which works on windows properly (even with
```
Comment (scopatz) [inline news/rmtree.rst]: This should be `xonsh.lib.os.rmtree()`

File: xonsh/lib/os.xsh [hunk 5]
```diff
@@ -9,3 +10,27 @@ def indir(d):
...
+    dirname : str
+        Directory to be removed
+    force : bool
+        If True force removal, defaults to False
+    """
+    try:
+        if force:
+            rm -rf @(dirname)
+        else:
+            rm -r @(dirname)
```
Comment (scopatz) [inline xonsh/lib/os.xsh]: Please move the argument logic outside of the try-block

**Comment by scopatz** [comment]
No need to squash!

**Comment by scopatz** [comment]
Thanks @CJ-Wright!

**Pull request merged**